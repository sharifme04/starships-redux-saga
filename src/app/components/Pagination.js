import React from "react";
import PropTypes from "prop-types";

const Pagination = props => {
  const { previousPage, currentPage, nextPage, starships } = props;
  return (
    <div className="d-flex justify-content-around mt-4">
      <div>
        <button
          type="button"
          className="btn btn-warning"
          disabled={starships && !starships.previous}
          onClick={() => previousPage(currentPage)}
        >
          Previous
        </button>
      </div>
      <div className="circle">
        <p>
          <b>Page: {currentPage}</b>
        </p>
      </div>
      <button
        type="button"
        className="btn btn-primary"
        disabled={starships && !starships.next}
        onClick={() => nextPage(currentPage)}
      >
        Next
      </button>
    </div>
  );
};

Pagination.propTypes = {
  currentPage: PropTypes.number,
  previousPage: PropTypes.func,
  nextPage: PropTypes.func,
  starships: PropTypes.object
};

export default Pagination;
