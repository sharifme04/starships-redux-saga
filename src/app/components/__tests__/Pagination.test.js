import React from "react";
import { shallow } from "enzyme";
import Pagination from "../Pagination";

describe("<Pagination />", () => {
  test("render", () => {
    const component = shallow(
      <Pagination nextPage={() => {}} previousPage={() => {}} currentPage={1} />
    );
    expect(component).toMatchSnapshot();
  });

  test("return the default when there is no data to map through", () => {
    const component = shallow(<Pagination />);
    expect(component).toMatchSnapshot();
  });

  test("does not break without value", () => {
    const component = shallow(<Pagination />);
    expect(component.find("button")).toHaveLength(2);
  });

  test("does not break with an empty array", () => {
    const component = shallow(<Pagination starships={{}} />);
    expect(component.find("button")).toHaveLength(2);
  });
});
