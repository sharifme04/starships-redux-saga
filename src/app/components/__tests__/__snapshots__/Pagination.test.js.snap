// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> render 1`] = `
<div
  className="d-flex justify-content-around mt-4"
>
  <div>
    <button
      className="btn btn-warning"
      onClick={[Function]}
      type="button"
    >
      Previous
    </button>
  </div>
  <div
    className="circle"
  >
    <p>
      <b>
        Page: 
        1
      </b>
    </p>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;

exports[`<Pagination /> return the default when there is no data to map through 1`] = `
<div
  className="d-flex justify-content-around mt-4"
>
  <div>
    <button
      className="btn btn-warning"
      onClick={[Function]}
      type="button"
    >
      Previous
    </button>
  </div>
  <div
    className="circle"
  >
    <p>
      <b>
        Page: 
      </b>
    </p>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
